src/bp-integrations/learndash/library/SyncGenerator.php

```diff
@@ -650,10 +650,17 @@ protected function createBuddypressGroup() {
 		$ldGroup  = get_post( $this->ldGroupId );
 		$settings = bp_ld_sync( 'settings' );
 
+		// Get the bp parent group id associate with ld parent group.
+		$bp_parent_group_id = 0;
+		if ( ! empty( $ldGroup->post_parent ) ) {
+			$bp_parent_group_id = get_post_meta( $ldGroup->post_parent, '_sync_group_id', true );
+		}
+
 		$this->bpGroupId = groups_create_group(
 			array(
 				'name'   => $ldGroup->post_title ?: "For Social Group: {$this->ldGroupId}",
 				'status' => $settings->get( 'learndash.default_bp_privacy' ),
+				'parent_id' => $bp_parent_group_id
```

Comment:
@surajkrsingh Follow the proper spacing here.

Reply:
@chetansatasiya changes done